dashboard/app/models/levels/level_group.rb

```diff
@@ -201,12 +201,8 @@ def self.get_survey_results(script, section)
   # @param {User} user - The optional user we're trying to see the attempt of
   # @param {Level} level - The sublevel we'd like the last attempt for
   def self.get_sublevel_last_attempt(current_user, user, level)
-    if user && current_user && user != current_user
-      # load other user's solution for teachers viewing their students' solution
-      user.last_attempt(level).try(:level_source).try(:data)
-    elsif current_user
-      # load current_user's previous attempt at this puzzle.
-      current_user.last_attempt(level).try(:level_source).try(:data)
-    end
+    # if given an alternative user, we want to show that user's solution (for
+    # teachers viewing students' solutions), otherwise show that of the current_user
+    (user || current_user).last_attempt(level).try(:level_source).try(:data)
```

Comment:
I think this refactor is incorrect - `(user || current_user)` could be `nil`, so this needs a `.try()` for `last_attempt`

Reply:
You are right. I'd looked at your original suggestion, decide for some reason that the first `try` you had was unnecessary, and left it out, but it's def needed. Thanks for catching.